### ISSUE TITLE
Paper Result 

### ISSUE THREAD

**Issue by piginhome** [sintel-dev/Orion#309]
* Orion version: 0.3.2
* Python version: python 3.7.13
* Operating System: linux

### Description

Describe what you were trying to get done.
Tell us what happened, what went wrong, and what you expected to happen.

After i written the paper, I would like to reproduce the result for NASA dataset or other dataset of TadGAN paper.
the below code is mine, but test result is not good. 
first of all, the reconstruction is not reproduce.  for example
training data:
![image](https://user-images.githubusercontent.com/61695020/178714312-1ac4aba8-1415-4096-90b5-8ae6a246067a.png)
test data and reconstruction:
![image](https://user-images.githubusercontent.com/61695020/178714419-de8324a7-b0c0-4b74-b460-eaf7257a5b8b.png)
result :
![image](https://user-images.githubusercontent.com/61695020/178714494-46a1c528-9c68-48cd-940b-5b8c5aba0dac.png)

So, i would like to receive the entire source code for TadGAN. if I cannot receive it, I hope that you modify the below code.
if you have any question, let me know. 

### What I Did

`#220712 testversion
from orion import Orion
from orion.primitives.tadgan import TadGAN
from orion.evaluation.contextual import contextual_confusion_matrix
import orion.data as od
import numpy as np
import pandas as pd
import os
import json
from orion.evaluation.contextual import contextual_accuracy, contextual_f1_score , contextual_recall, contextual_precision

metrics = [
    'f1',
    'recall',
    'precision',
]
SIGNALS_SMAP = (
'P-1', 'S-1', 'E-1', 'E-2', 'E-3', 'E-4', 'E-5', 'E-6', 'E-7',
	'E-8', 'E-9', 'E-10', 'E-11', 'E-12', 'E-13', 'A-1', 'D-1', 'P-3',
	'D-2', 'D-3', 'D-4', 'A-2', 'A-3', 'A-4', 'G-1', 'G-2', 'D-5',
	'D-6', 'D-7', 'F-1', 'P-4', 'G-3', 'T-1', 'T-2', 'D-8', 'D-9',
	'F-2', 'G-4', 'T-3', 'D-11', 'D-12', 'B-1', 'G-6', 'G-7', 'P-7',
	'R-1', 'A-5', 'A-6', 'A-7', 'D-13', 'A-8', 'A-9', 'F-3'
)

SIGNALS_SML = (
'M-6', 'M-1', 'M-2', 'S-2', 'P-10', 'T-4', 'T-5', 'F-7', 'M-3', 'M-4', 
	'M-5', 'P-15', 'C-1', 'C-2', 'T-12', 'T-13', 'F-4', 'F-5', 'D-14', 
	'T-9', 'P-14', 'T-8', 'P-11', 'D-15', 'D-16', 'M-7', 'F-8'
)

DATAPATH = 'csv/'

	#hyperparameters
	#pipeline training

#with open('./Orion/orion/pipelines/verified/tadgan/tadgan_msl.json') as f:
#	hyperparameters = json.load(f)#

results = np.zeros([len(SIGNALS_SML), 4]) #[tn, fp, fn, tp]

for i, signal in enumerate(SIGNALS_SML):

	#orion = Orion(pipeline ='tadgan', hyperparameters = hyperparameters)
	
	train_data = od.load_signal(DATAPATH + signal +'-train.csv')
	test_data = od.load_signal(DATAPATH + signal +'-test.csv')
	true_anomalies = od.load_anomalies(signal)
	
	X, index = time_segments_aggregate(train_data, interval=21600, time_column='timestamp')
	X_test, test_index = time_segments_aggregate(test_data, interval=21600, time_column='timestamp')
	#data imputation
	imp = SimpleImputer()
	X = imp.fit_transform(X)
	X_test = imp.fit_transform(X_test)

	#data normalization
	scaler = MinMaxScaler(feature_range=(-1, 1))
	X = scaler.fit_transform(X)
	X_test= scaler.fit_transform(X_test)

	plot_ts(X)
	X_test, y_test, X_test_index, y_test_index = rolling_window_sequences(X_test, test_index, 
													window_size=100, 
													target_size=1, 
													step_size=1,
													target_column=0)
	#rollin window
	X, y, X_index, y_index = rolling_window_sequences(X, index, 
													window_size=100, 
													target_size=1, 
													step_size=1,
													target_column=0)

	print("Training data input shape: {}".format(X.shape))
	print("Training data index shape: {}".format(X_index.shape))
	print("Training y shape: {}".format(y.shape))
	print("Training y index shape: {}".format(y_index.shape))
	print("test data input shape: {}".format(X_test.shape))
	print("test data index shape: {}".format(X_test_index.shape))
	print("test y shape: {}".format(y_test.shape))
	print("test y index shape: {}".format(y_test_index.shape))
	#hyperparameters
	#pipeline training

	hyperparameters["epochs"] = 70
	hyperparameters["input_shape"] = (100, 1) # based on the window size
	hyperparameters["optimizer"] = "keras.optimizers.Adam"
	hyperparameters["learning_rate"] = 0.0005
	hyperparameters["latent_dim"] = 20
	hyperparameters["batch_size"] = 64
	tgan = TadGAN(**hyperparameters)
	tgan.fit(X)
	# reconstruct
	X_hat, critic = tgan.predict(X_test)

	# visualize X_hat
	plot_rws(X_hat)

	# flatten the predicted windows
	y_hat = unroll_ts(X_hat)
	test111 = y_hat.reshape(len(y_test),1)
	# plot the time series
	plot_ts([y_test, y_hat], labels=['original', 'reconstructed'])

	# pair-wise error calculation
	error = np.zeros(shape=y_test.shape)
	length = y_test.shape[0]
	for i in range(length):
		error[i] = abs(y_hat[i] - y_test[i])

	# visualize the error curve
	fig = plt.figure(figsize=(30, 3))
	plt.plot(error)
	plt.show()

	#score anomalies
	#error, true_index, true, pred = score_anomalies(X_test, X_hat, critic, X_test_index, rec_error_type="dtw", comb="mult")
	error, true_index, true, pred = score_anomalies(y_test, test111, critic, y_test_index, rec_error_type="dtw", comb="mult")
	pred = np.array(pred).mean(axis=2)

	# visualize the error curve
	plot_error([[true, pred], error])
	# find anomalies
	intervals = find_anomalies(error, y_test_index, 
							window_size_portion=0.33, 
							window_step_size_portion=0.1, 
							fixed_threshold=True)
	print(intervals)

	# visualize the result
	anomalies = pd.DataFrame(intervals, columns=['start', 'end', 'score'])
	plot(test_data, [anomalies, true_anomalies])
	accuracy = contextual_accuracy(true_anomalies, anomalies, start=X_test_index[0], end=X_test_index[-1], weighted=True)
	f1_score = contextual_f1_score(true_anomalies, anomalies, start=X_test_index[0], end=X_test_index[-1], weighted=True)
	recall = contextual_recall(true_anomalies, anomalies, start=X_test_index[0], end=X_test_index[-1], weighted=True)
	precision = contextual_precision(true_anomalies, anomalies, start=X_test_index[0], end=X_test_index[-1], weighted=True)

	print("Accuracy score = {:0.3f}".format(accuracy))
	print("F1 score = {:0.3f}".format(f1_score))
	print("recall = {:0.3f}".format(recall))
	print("precision = {:0.3f}".format(precision))

```
Paste the command(s) you ran and the output.
If there was a crash, please include the traceback here.
```

**Comment by sarahmish** [comment]
Hi @piginhome, thank you for using Orion!

If you would like to reproduce the benchmark results for the NASA dataset, you can do so by running our benchmark framework! Here's a snippet to get you started:

```python3
from orion.benchmark import benchmark, BENCHMARK_DATA

datasets = {
    "MSL": BENCHMARK_DATA["MSL"],
    "SMAP": BENCHMARK_DATA["SMAP"]
}
pipelines = {"tadgan": "tadgan"}

scores = benchmark(pipelines=pipelines, datasets=datasets)
```
